### PR TITLE
[RDBMS] `az postgres flexible-server update-index-tuning` and `az postgres flexible-server tuning-options`: Support enabling Index tuning

### PR DESCRIPTION
**Pull request by nasc17**
**Related command**
`az postgres flexible-server update-index-tuning`
`az postgres flexible-server tuning-options`

**Description**
Adding cli support of Index tuning to analyzes read queries captured in Query Store and recommends index changes to optimize these queries.

**Testing Guide**
Manually and azdev

**History Notes**

[RDBMS] `az postgres flexible-server update-index-tuning` and `az postgres flexible-server tuning-options`: Support enabling Index tuning to analyzes read queries captured in Query Store and recommends index changes to optimize these queries

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py [hunk 1]
```diff
@@ -286,3 +292,12 @@ def load_flexibleserver_command_table(self, _):
                             client_factory=cf_postgres_flexible_private_link_resources) as g:
         g.command('list', 'list_by_server')
         g.custom_show_command('show', 'flexible_server_private_link_resource_get', custom_command_type=flexible_servers_custom_postgres)
+
+    with self.command_group('postgres flexible-server index-tuning', postgres_flexible_config_sdk,
+                            custom_command_type=flexible_servers_custom_postgres,
+                            client_factory=cf_postgres_flexible_config, table_transformer=table_transform_output_parameters) as g:
+        g.custom_command('update', 'index_tuning_update', custom_command_type=flexible_servers_custom_postgres)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:299]: If there's only one command for a command group, then we don't suggest to make it a command group. This case for example can be defined as `az postgres flexible-server update-index-tuning`

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py [hunk 2]
```diff
@@ -286,3 +292,12 @@ def load_flexibleserver_command_table(self, _):
...
+
+    with self.command_group('postgres flexible-server tuning-options', postgres_flexible_tuning_options_sdk,
+                            client_factory=cf_postgres_flexible_tuning_options) as g:
+        g.custom_command('show', 'tuning_options_get', custom_command_type=flexible_servers_custom_postgres)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:303]: Same here. Can be `az postgres flexible-server show-tuning-options`

Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:303]: Same here. Can be `az postgres flexible-server show-tuning-options`

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py (hunk 1 above)
Comment (nasc17) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py]: Done.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py (hunk 2 above)
Comment (nasc17) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:303]: More commands to come for tuning-options

**Comment by nasc17** [comment]
`az postgres flexible-server tuning-options` will have more commands to come in future releases

**Pull request closed without merging**